File: bindings/dart/example/browser_relay.dart
import 'package:maxt/maxt.dart';

const _relayUrl = String.fromEnvironment('MAXT_RELAY_URL');

Future<void> main() async {
  // A public read works directly in a browser. A relay URL is only needed when
  // configuring credentials for signed requests.
  await Maxt.initialize(relayUrl: _relayUrl.isEmpty ? null : _relayUrl);
  try {
    final market = Market.spot(Exchange.binance, 'BTC', 'USDT');
    final ticker = await Client(BinanceAdapter.spot()).ticker(market);
    print('${ticker.market}: ${ticker.lastPrice}');
    print(
      'For signed browser calls, use a trusted relay and explicit opt-in; see relay/README.md.',
    );
  } finally {
    await Maxt.dispose();
  }
}

File: bindings/typescript/examples/browser-relay.mjs
// Browser entry point: public reads connect directly. Signed calls need a relay.
import {
  BinanceAdapter,
  Client,
  Exchange,
  Market,
  initialize,
} from "@jabdori/maxt/browser";

// Set this to a trusted relay URL only for signed browser calls.
const relayUrl = null;
await initialize(
  relayUrl === null
    ? {}
    : { relayUrl, allowInsecureBrowserCredentials: true },
);

const market = Market.spot(Exchange.Binance, "BTC", "USDT");
const ticker = await new Client(BinanceAdapter.spot()).ticker(market);
console.log(`${ticker.market}: ${ticker.lastPrice}`);
console.log("For signed browser calls, set a trusted relay URL and use the deployment controls in relay/README.md.");
